### ISSUE TITLE
flake8 3.7.0+ fails with "ImportError: cannot import name 'Type'" in python 3.5.0 and 3.5.1

### ISSUE THREAD

**Issue by asottile** [PyCQA/flake8#674]
In GitLab by @breetz on Jan 30, 2019, 13:44

```
$ pip install flake8
```

I can't run `flake8 --bug-report` since I can't run it at all.

Using my system python3.5 installation of 3.5.1, I cannot run flake8. It fails with an ImportError:
```
Traceback (most recent call last):
  File ".tox/py35/bin/flake8", line 7, in <module>
    from flake8.main.cli import main
  File "/nail/home/bariteau/pg/yelp_compose/.tox/py35/lib/python3.5/site-packages/flake8/main/cli.py", line 4, in <module>
    from flake8.main import application
  File "/nail/home/bariteau/pg/yelp_compose/.tox/py35/lib/python3.5/site-packages/flake8/main/application.py", line 8, in <module>
    from typing import Type, TYPE_CHECKING  # noqa: F401 (until flake8 3.7)
ImportError: cannot import name 'Type'
```

Here's some relevant commands:
```
$ python --version
Python 3.5.1
$ flake8
Traceback (most recent call last):
  File "/nail/home/bariteau/foo/venv/bin/flake8", line 6, in <module>
    from flake8.main.cli import main
  File "/nail/home/bariteau/foo/venv/lib/python3.5/site-packages/flake8/main/cli.py", line 4, in <module>
    from flake8.main import application
  File "/nail/home/bariteau/foo/venv/lib/python3.5/site-packages/flake8/main/application.py", line 8, in <module>
    from typing import Type, TYPE_CHECKING  # noqa: F401 (until flake8 3.7)
ImportError: cannot import name 'Type'
```

Quick investigation show that typing.Type was added in 3.5.2 https://docs.python.org/3.5/whatsnew/changelog.html#id24

**Comment by asottile** [comment]
In GitLab by @breetz on Jan 30, 2019, 13:44

closed

**Comment by asottile** [comment]
In GitLab by @breetz on Jan 30, 2019, 13:44

reopened

**Comment by asottile** [comment]
In GitLab by @breetz on Jan 30, 2019, 13:44

changed title from **flake8 3.7.0+ fails with "ImportError: cannot import name 'Type'" in python 3.5.0 {-or-} 3.5.1** to **flake8 3.7.0+ fails with "ImportError: cannot import name 'Type'" in python 3.5.0 {+and+} 3.5.1**

**Comment by asottile** [comment]
In GitLab by @breetz on Jan 30, 2019, 13:44

changed the description

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 13:48

urgh, I'll remove that type annotation -- we aren't using mypy completely in this repo yet anyway.  shame that they introduced it in a patch version :( (though I guess that's what you get for a provisional module)

just curious, which system ships with 3.5.0 / 3.5.1? (2015-12-06)

**Comment by asottile** [comment]
In GitLab by @breetz on Jan 30, 2019, 13:52

I'm running this from a Ubuntu 14.04 (Trusty Tahr) box. I'm not entirely sure of the provenance.

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 13:53

oh, yeah that's probably _my_ backport from when I was there :laughing: 

looks like the `typing` module in 3.5.0 / 3.5.1 is really lacking (scrolling through the other things that are also missing) -- let me build an old version of python and get flake8 importing :S

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 14:06

mentioned in merge request !294

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 14:20

closed via merge request !294

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 14:20

mentioned in commit e21d74fcc4dfbaf36aaa418215ad0615793f14fc

**Comment by asottile** [comment]
In GitLab by @asottile on Jan 30, 2019, 19:59

This should be fixed in 3.7.3 :tada:

thanks again for the issue, friend
